odl/deform/linearized_deform.py

```diff
@@ -0,0 +1,538 @@
+
+    and the vector field ``v`` is a displacement mapping:
+
+        y --> v(y)
+
+    Here, the ``y`` is an element in the domain of the target.
+    In diffeomorphic sense, the domain of ``I`` is the same
+    as the target.
+    """
+
+    def __init__(self, displacement):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        displacement : `ProductSpaceVector`
+            Fixed displacement field used in the linearized deformation.
+
+        Examples
+        --------
+        >>> import odl
+        >>> space = odl.uniform_discr(0, 1, 5)
+        >>> disp_field_space = odl.ProductSpace(space, space.ndim)
+        >>> displacement_field = disp_field_space.element([[0, 0, 0, -0.2, 0]])
+        >>> op = LinDeformFixedDisp(displacement_field)
+        >>> template = space.element([0, 0, 1, 0, 0])
+        >>> op(template)
+        uniform_discr(0.0, 1.0, 5).element([0.0, 0.0, 1.0, 1.0, 0.0])
+        """
+        if not isinstance(displacement.space, odl.ProductSpace):
+            raise TypeError('`displacement` {!r} not an element'
+                            'of `ProductSpace`'.format(displacement))
+        if not isinstance(displacement.space[0], odl.DiscreteLp):
+            raise TypeError('`displacement[0]` {!r} not an element of'
+                            '`DiscreteLp`'.format(displacement[0]))
+
+        domain_space = displacement[0].space
+
+        Operator.__init__(self, domain_space, domain_space, linear=True)
+
+        self.displacement = displacement
+
+    def _call(self, template):
+        """Implementation of ``self(template)``.
+
+        Parameters
+        ----------
+        template : `domain` element
+            Given template that is to be deformed by the fixed
+            displacement field.
+        """
+        return linear_deform(template, self.displacement)
+
+    @property
+    def adjoint(self):
+        """Adjoint of the operator.
+
+        Returns
+        -------
+        adj_op : `LinDeformFixedDispAdj`
+            The adjoint of the operator.
+        """
+        return LinDeformFixedDispAdj(self.displacement)
+
+
+class LinDeformFixedDispAdj(Operator):
+
+    """Adjoint of the fixed displacement linearized deformation operator.
+
+    This operator computes the adjoint of the linear operator that
+    map a template ``I`` to its deformation using a fixed displacement ``v``:
+
+        I --> exp(-div(v)) * I(Id - v)
+
+    Here, the ``I`` is the given template, ``Id`` is the identity mapping:
+
+        y --> y
+
+    the vector field ``v`` is a displacement field:
+
+        y --> v(y)
+
+    and, ``exp(-div(v))`` is an approximation of the determinant of the
+    Jacobian of ``(Id + v)^{-1}``, which is accurate in some sense if the
+    magnitude of``v`` is close to ``0``.
+    """
+
+    def __init__(self, displacement):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        displacement : `ProductSpaceVector`
```

Comment:
I've mentioned this several times, but the problem with this formulation is that it is far from sufficient. All of the following spaces would fail:

```python
ProductSpace(rn(2), 3)
ProductSpace(rn(2), rn(3))
ProductSpace(uniform_discr(0, 1, 5), rn(3))
```

We need to make sure that the user knows that these will not work. That's why `ProductSpaceVector` is not sufficient.